src/server/game/Spells/SpellInfo.cpp

```diff
@@ -1473,16 +1473,43 @@ SpellCastResult SpellInfo::CheckLocation(uint32 map_id, uint32 zone_id, uint32 a
     }
 
     // continent limitation (virtual continent)
-    if (AttributesEx4 & SPELL_ATTR4_CAST_ONLY_IN_OUTLAND)
+    if (HasAttribute(SPELL_ATTR4_CAST_ONLY_IN_OUTLAND))
     {
         uint32 v_map = GetVirtualMapForMapAndZone(map_id, zone_id);
         MapEntry const* mapEntry = sMapStore.LookupEntry(v_map);
         if (!mapEntry || mapEntry->addon < 1 || !mapEntry->IsContinent())
             return SPELL_FAILED_INCORRECT_AREA;
     }
 
+    // continent limitation (virtual continent)
+    if (HasAttribute(SPELL_ATTR4_CAST_ONLY_IN_OUTLAND))
+    {
+        if (strict)
+        {
+            AreaTableEntry const* areaEntry = sAreaTableStore.LookupEntry(area_id);
+            if (!areaEntry)
+            {
+                areaEntry = sAreaTableStore.LookupEntry(zone_id);
+            }
+
+            if (!areaEntry || !areaEntry->IsFlyable() || !player->canFlyInZone(map_id, zone_id, this))
+            {
+                return SPELL_FAILED_INCORRECT_AREA;
```

Comment:
you could remove the condition `!areaEntry` and add the return even in the previous if conditions, but it is not so relevant, so your choice
(you only save one double check)

Reply:
Null check